camera_control/camera_control/nodes/camera_control_node.py

```diff
@@ -0,0 +1,264 @@
+""" File realsense.py
+
+Class for using a realsense depth camera. 
+"""
+
+import cv2
+import rclpy
+import time
+from datetime import datetime
+from rclpy.node import Node
+
+from camera_control.camera.realsense import RealSense
+from camera_control_interfaces.msg import (
+    PosData,
+    Pos,
+    Point,
+    UInt8NumpyArray,
+    Int16NumpyArray,
+)
+from camera_control_interfaces.srv import GetFrame, GetPosData, SendEnable
+
+import camera_control.utils.cv_utils as cv_utils
+from ultralytics import YOLO
+
+import numpy as np
+import os
+
+from ament_index_python.packages import get_package_share_directory
+
+
+class CameraControlNode(Node):
+    def __init__(self):
+        Node.__init__(self, "CameraControlNode")
+        self.logger = self.get_logger()
+
+        # declare parameters from a ros config file, if no parameter is found, the default is used
+        self.declare_parameters(
+            namespace="",
+            parameters=[
+                ("video_dir", "/opt/video_stream"),
+                ("debug_video_dir", "/opt/debug_video_stream"),
+                ("save_video", True),
+                ("save_debug", False),
+                ("frame_period", 0.1),
+                ("rgb_size", [1920, 1080]),
+                ("depth_size", [1280, 720]),
+            ],
+        )
+
+        # get class attributes from passed in parameters
+        self.video_dir = (
+            self.get_parameter("video_dir").get_parameter_value().string_value
+        )
+        self.debug_video_dir = (
+            self.get_parameter("debug_video_dir").get_parameter_value().string_value
+        )
+        self.rec_video_frame = (
+            self.get_parameter("save_video").get_parameter_value().bool_value
+        )
+        self.rec_debug_frame = (
+            self.get_parameter("save_debug").get_parameter_value().bool_value
+        )
+        self.frame_period = (
+            self.get_parameter("frame_period").get_parameter_value().double_value
```

Comment:
might be easier to define as FPS integer value

Reply:
I'll change it over when I move to using it in the main node